pikapkg/snowpack
# Test fixtures continued

## Changes

Continues work done on #2858 by migrating tests from `/build` to `/snowpack` and using the snowpack API over the CLI to verify correct behaviours. Currently working through all the tests one by one and converting them.

Currently converted:

- [x] Named imports
- [x] CDN Url tests
- [x] Config alias
- [x] Config metaUrlPath
- [x] Extends plugins
- [x] Config external packages
- [x] Plugins instantiated object
- [x] Config metaUrlPath with mounted dir
- [x] Config mount (basic and advanced)
- [x] Config optimize test
- [x] Config out test
- [x] Config package lookup fields test
- [x] Optimize treeshake test
- [x] Module resolution test
- [x] Css modules test
- [x] Another metaUrlPath test
- [x] Config invalid test
- [x] Html env variables test
- [x] Import dot folder test
- [x] Glob test
- [x] Import json test
- [x] Import ts test
- [x] Jsx inject test
- [x] Load config from cjs
- [ ] Load config from esm
- [ ] Load config from mjs
- [x] Load config from json
- [x] Load config from package json
- [x] Package bootstrap test
- [x] Source remote-next test
- [x] Package tippy test
- [ ] Package workspace test
- [x] Plugin build script test
- [x] Plugin svelte test
- [x] Plugin hook transform test
- [ ] Plugin run script test
- [x] Plugin sass test
- [x] Plugin vue test
- [x] Module resolve js test
- [x] Scan imports html test
- [x] Utf8 test

General changes to the tests generally or test fixture script:

- Make test runner use loadConfiguration
- Adds snowpack cache to test result output
- Set logger level to error to clean up test output
- Use toContain consistently
- Try using readFileSync to prevent CI failure

Changes made to Snowpack itself

- Make loadConfiguration check for files in provided root

## Testing

These are the tests 😅 

## Docs

I created a very simple template from the examples of some new tests that already existed:

```js
const {testFixture} = require('../../fixture-utils');
const dedent = require('dedent');

describe('suite', () => {
  beforeAll(() => {
    // Needed until we make Snowpack's JS Build Interface quiet by default
    require('snowpack').logger.level = 'warn';
  });

  it('tests', async () => {
    const result = await testFixture(
      {},
      {
        'index.js': dedent``,
      },
    );
    expect(result['index.js']).toMatchSnapshot();
  });
});

```
